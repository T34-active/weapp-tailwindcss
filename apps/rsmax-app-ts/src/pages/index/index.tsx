import { Image, Text, View } from 'rsmax/one'
import styles from './index.css'

export default () => {
  return (
    <View className={styles.app}>
      <View className={styles.header}>
        <Image
          src="https://gw.alipayobjects.com/mdn/rms_b5fcc5/afts/img/A*OGyZSI087zkAAAAAAAAAAABkARQnAQ"
          className={styles.logo}
        />
        <View className={styles.text}>
          编辑
          {' '}
          <Text>src/pages/index/index.js</Text>
          开始
        </View>
      </View>
    </View>
  )
}

// TypeError: Cannot read property 'current' of undefined
//     at li.callLifecycle (createPageConfig.js:142)
//     at li.onLoad (createPageConfig.js:100)
//     at l.<anonymous> (VM496 WASubContext.js:1)
//     at VM496 WASubContext.js:1
//     at X (VM496 WASubContext.js:1)
//     at VM496 WASubContext.js:1
//     at Array.forEach (<anonymous>)
//     at VM496 WASubContext.js:1
//     at Array.forEach (<anonymous>)
//     at VM496 WASubContext.js:1(env: macOS,mp,1.06.2503300; lib: 3.8.7)
