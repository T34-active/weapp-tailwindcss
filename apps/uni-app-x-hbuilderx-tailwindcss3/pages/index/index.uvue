<template>
	<view class="px-4">
		<view class="mt-[32.43rpx] bg-[#322323]">
			<text class="text-[#844343]">{{title}}</text>
		</view>
	</view>
</template>

<script>
	export default {
		data() {
			return {
				title: 'Hello'
			}
		},
		onLoad() {

		},
		methods: {

		}
	}
</script>

<style>

</style>