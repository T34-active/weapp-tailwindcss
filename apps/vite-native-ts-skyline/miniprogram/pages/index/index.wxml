<!-- index.wxml -->
<navigation-bar title="Weixin" back="{{false}}" color="black" background="#FFF"></navigation-bar>
<view class="min-h-screen bg-slate-950">
  <view class="p-8">
    <view class="i-mdi-home bg-[#43ffff] text-[200px]"></view>
    <view class="usermotto">
      <text class="user-motto text-[#234fff] bg-[#43ffff]">{{motto}}</text>
    </view>
  </view>
</view>