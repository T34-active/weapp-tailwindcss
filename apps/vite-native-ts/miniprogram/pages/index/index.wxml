<view class="p-4">
  <IceButton class="bg-[#d72929]" tw-class="bg-[orange]" hover-class="bg-[green]" tw-hover-class="bg-[pink]"></IceButton>
</view>