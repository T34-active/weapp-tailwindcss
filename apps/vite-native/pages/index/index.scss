// https://sass-lang.com/documentation/at-rules/import/#plain-css-imports
/* stylelint-disable-next-line import-notation */
// @import url('../../app.scss');

// @weapp-vite-keep-import 'xx';

// @use '../../app.wxss';

.s {
  .a {
    color: turquoise;
  }
}

.user-motto {
  font-size: 12px;
}
