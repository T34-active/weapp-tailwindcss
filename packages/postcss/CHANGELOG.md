# @weapp-tailwindcss/postcss

## 1.0.10-alpha.1

### Patch Changes

- [`23c08c7`](https://github.com/sonofmagic/weapp-tailwindcss/commit/23c08c7d7ad276035ffa3ef78d0fb00e1c717ddf) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: isTailwindcssV4ModernCheck

## 1.0.10-alpha.0

### Patch Changes

- [`0d6564a`](https://github.com/sonofmagic/weapp-tailwindcss/commit/0d6564aaaea588289dac5b1b784b1902105b297a) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: 优化 isTailwindcssV4ModernCheck 判断

## 1.0.9

### Patch Changes

- [`3113053`](https://github.com/sonofmagic/weapp-tailwindcss/commit/31130538a13098e2a1d29bbd331dc67a195689cf) Thanks [@sonofmagic](https://github.com/sonofmagic)! - Fix: Support `Tailwindcss@4.1.1` and fix [#619](https://github.com/sonofmagic/weapp-tailwindcss/issues/619)

## 1.0.8

### Patch Changes

- [`e8c4534`](https://github.com/sonofmagic/weapp-tailwindcss/commit/e8c4534ab588eae1115dac15b529b7a122141316) Thanks [@sonofmagic](https://github.com/sonofmagic)! - feat: 添加 `cssRemoveProperty` 选项，默认值为 `true`，这是为了在 `tailwindcss` 中移除这种 css 节点:

  ```css
  @property --tw-content {
    syntax: "*";
    initial-value: "";
    inherits: false;
  }
  ```

  这种样式在小程序中，没有任何的意义。

## 1.0.7

### Patch Changes

- [`e0c37f5`](https://github.com/sonofmagic/weapp-tailwindcss/commit/e0c37f5f546b143341a75701a1907f876df38fa9) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore(deps): upgrade

- Updated dependencies [[`e0c37f5`](https://github.com/sonofmagic/weapp-tailwindcss/commit/e0c37f5f546b143341a75701a1907f876df38fa9)]:
  - @weapp-tailwindcss/shared@1.0.2

## 1.0.7-alpha.0

### Patch Changes

- [`e0c37f5`](https://github.com/sonofmagic/weapp-tailwindcss/commit/e0c37f5f546b143341a75701a1907f876df38fa9) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore(deps): upgrade

- Updated dependencies [[`e0c37f5`](https://github.com/sonofmagic/weapp-tailwindcss/commit/e0c37f5f546b143341a75701a1907f876df38fa9)]:
  - @weapp-tailwindcss/shared@1.0.2-alpha.0

## 1.0.6

### Patch Changes

- [`ff9933a`](https://github.com/sonofmagic/weapp-tailwindcss/commit/ff9933ad06de7bf3333c1c63016920639a56b87a) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: tailwindcss@4 space-x-_ and space-y-_

## 1.0.5

### Patch Changes

- [`a4532ab`](https://github.com/sonofmagic/weapp-tailwindcss/commit/a4532ab34de62556e57ed350e15ca14e602b7f93) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: 使用space-y-2后编译报错 #595

## 1.0.4

### Patch Changes

- [`9e65534`](https://github.com/sonofmagic/weapp-tailwindcss/commit/9e65534f035ee4e17a2dc0b891278cacb92d5a0b) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: @layer as :not(#n) and @weapp-tailwindcss/merge switcher

## 1.0.3

### Patch Changes

- Updated dependencies [[`bdbca26`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bdbca268738ef033a5789e8a6713608c4d599b61)]:
  - @weapp-tailwindcss/shared@1.0.1

## 1.0.3-alpha.0

### Patch Changes

- Updated dependencies [[`bdbca26`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bdbca268738ef033a5789e8a6713608c4d599b61)]:
  - @weapp-tailwindcss/shared@1.0.1-alpha.0

## 1.0.2

### Patch Changes

- [`64c0189`](https://github.com/sonofmagic/weapp-tailwindcss/commit/64c018935732481ebe2f366e4136b4d3574dde57) Thanks [@sonofmagic](https://github.com/sonofmagic)! - feat: improve `isAllowedClassName` preflight

## 1.0.1

### Patch Changes

- [`ee34fb3`](https://github.com/sonofmagic/weapp-tailwindcss/commit/ee34fb34688a2bd11018ce5e4ea6d07a062b0b55) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: bg-gradient --tw-gradient-position in oklab

## 1.0.0

### Major Changes

- [`bafd149`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bafd149f0510b30cbf95711223583055023e7875) Thanks [@sonofmagic](https://github.com/sonofmagic)! - ## Feature

  增加 `@weapp-tailwindcss/merge` 支持，这是 `weapp-tailwindcss` 版本的 `tailwindcss-merge` 和 `cva` 方法

  ## Breaking Changes

  1. 去除 `weapp-tailwindcss/postcss` (可直接安装使用 `@weapp-tailwindcss/postcss`)
  2. 增加 `weapp-tailwindcss/escape` 来取代 `weapp-tailwindcss/replace`
  3. 项目 monorepo 区分包

### Patch Changes

- [`d4bfcc7`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d4bfcc7a25776eb9869bd934c05e3b49a3f4cc8b) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: remove attribute in :where pseudo

- [`2c51531`](https://github.com/sonofmagic/weapp-tailwindcss/commit/2c515310f1fdfd15d11e2e35213c7e6bfcb55c3d) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore(deps): upgrade

- [`d7fa028`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d7fa02877ce74792687765766ff94ae3e30edf3b) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: rename export function

- [`1d689ef`](https://github.com/sonofmagic/weapp-tailwindcss/commit/1d689efca6cf0de7e476b03b2be8d09284beae68) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: change postcssPresetEnv default value

- [`1df699b`](https://github.com/sonofmagic/weapp-tailwindcss/commit/1df699bf82d66847cbc64cc8f294ef237e0470c5) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: fix types import intel

- [`6d20f1f`](https://github.com/sonofmagic/weapp-tailwindcss/commit/6d20f1f9a799cf350dcbbd861907f0ff70a68dfd) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: perfer postcss-preset-env instead of @csstools

- [`e745ba5`](https://github.com/sonofmagic/weapp-tailwindcss/commit/e745ba5cd9e232c0b5b7053538beb0772240eab8) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: bump version

- Updated dependencies [[`bafd149`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bafd149f0510b30cbf95711223583055023e7875), [`2c51531`](https://github.com/sonofmagic/weapp-tailwindcss/commit/2c515310f1fdfd15d11e2e35213c7e6bfcb55c3d), [`d7fa028`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d7fa02877ce74792687765766ff94ae3e30edf3b), [`d50d95d`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d50d95d04e1c6b7c3ff32acc0d9894d3c0f06d22), [`e745ba5`](https://github.com/sonofmagic/weapp-tailwindcss/commit/e745ba5cd9e232c0b5b7053538beb0772240eab8)]:
  - @weapp-tailwindcss/shared@1.0.0

## 1.0.0-alpha.8

### Patch Changes

- [`1d689ef`](https://github.com/sonofmagic/weapp-tailwindcss/commit/1d689efca6cf0de7e476b03b2be8d09284beae68) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: change postcssPresetEnv default value

## 1.0.0-alpha.7

### Patch Changes

- [`d4bfcc7`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d4bfcc7a25776eb9869bd934c05e3b49a3f4cc8b) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: remove attribute in :where pseudo

## 1.0.0-alpha.6

### Patch Changes

- [`e745ba5`](https://github.com/sonofmagic/weapp-tailwindcss/commit/e745ba5cd9e232c0b5b7053538beb0772240eab8) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: bump version

- Updated dependencies [[`e745ba5`](https://github.com/sonofmagic/weapp-tailwindcss/commit/e745ba5cd9e232c0b5b7053538beb0772240eab8)]:
  - @weapp-tailwindcss/shared@1.0.0-alpha.4

## 1.0.0-alpha.5

### Patch Changes

- [`6d20f1f`](https://github.com/sonofmagic/weapp-tailwindcss/commit/6d20f1f9a799cf350dcbbd861907f0ff70a68dfd) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: perfer postcss-preset-env instead of @csstools

## 1.0.0-alpha.4

### Patch Changes

- [`d7fa028`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d7fa02877ce74792687765766ff94ae3e30edf3b) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: rename export function

- Updated dependencies [[`d7fa028`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d7fa02877ce74792687765766ff94ae3e30edf3b)]:
  - @weapp-tailwindcss/shared@1.0.0-alpha.3

## 1.0.0-alpha.3

### Patch Changes

- [`2c51531`](https://github.com/sonofmagic/weapp-tailwindcss/commit/2c515310f1fdfd15d11e2e35213c7e6bfcb55c3d) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore(deps): upgrade

- Updated dependencies [[`2c51531`](https://github.com/sonofmagic/weapp-tailwindcss/commit/2c515310f1fdfd15d11e2e35213c7e6bfcb55c3d)]:
  - @weapp-tailwindcss/shared@1.0.0-alpha.2

## 1.0.0-alpha.2

### Patch Changes

- [`1df699b`](https://github.com/sonofmagic/weapp-tailwindcss/commit/1df699bf82d66847cbc64cc8f294ef237e0470c5) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: fix types import intel

## 1.0.0-alpha.1

### Major Changes

- [`bafd149`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bafd149f0510b30cbf95711223583055023e7875) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: monorepo changes

### Patch Changes

- Updated dependencies [[`bafd149`](https://github.com/sonofmagic/weapp-tailwindcss/commit/bafd149f0510b30cbf95711223583055023e7875)]:
  - @weapp-tailwindcss/shared@1.0.0-alpha.1

## 0.0.1-alpha.0

### Patch Changes

- Updated dependencies [[`d50d95d`](https://github.com/sonofmagic/weapp-tailwindcss/commit/d50d95d04e1c6b7c3ff32acc0d9894d3c0f06d22)]:
  - @weapp-tailwindcss/shared@0.0.1-alpha.0
