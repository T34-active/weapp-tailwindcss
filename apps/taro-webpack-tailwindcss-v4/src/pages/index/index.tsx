import { Text, View } from '@tarojs/components'
import { useLoad } from '@tarojs/taro'
import './index.css'

export default function Index() {
  useLoad(() => {
    console.log('Page loaded.')
  })

  return (
    <>
      <View className="bg-[#2e2bcc] text-[100rpx] text-white">
        <Text>Hello world!</Text>
      </View>
      <View className="bg-purple-800 text-pink-200">
        11
      </View>
      <View className="h-14 bg-gradient-to-r from-cyan-500 to-blue-500"></View>
      <View className="bg-purple-800 text-pink-200 invert">
        invert
      </View>
      <View className="bg-purple-800 text-pink-200 backdrop-blur-2xl">
        backdrop-blur-2xl
      </View>
      <View className="skew-x-0 bg-purple-800 text-pink-200">
        skew-x-0
      </View>
    </>

  )
}
