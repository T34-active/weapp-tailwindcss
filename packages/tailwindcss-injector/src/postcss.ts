import type { PluginCreator } from 'postcss'
import type { Config } from 'tailwindcss'
import type { Options } from './types'
import postcss from 'postcss'
import set from 'set-value'
import tailwindcss from 'tailwindcss'
import { loadConfig } from 'tailwindcss-config'
import { getConfig } from './config'
import { postcssPlugin } from './constants'
import { regExpTest, removeFileExtension } from './utils'
import { getDepFiles } from './wxml'

// function isObject(obj: any): obj is object {
//   return typeof obj === 'object' && obj !== null
// }
export type {
  Options,
}

const creator: PluginCreator<Partial<Options>> = (options) => {
  const { config, filter, directiveParams, cwd, extensions, insertAfterAtRulesNames, insertAfterComments } = getConfig(options)

  const extensionsGlob = `{${extensions.join(',')}}`

  return {
    postcssPlugin,
    plugins: [
      {
        postcssPlugin: `${postcssPlugin}:post`,
        async Once(root, helpers) {
          if (filter(root.source?.input)) {
            const atruleLastIndex = root.nodes.findLastIndex(x => x.type === 'atrule' && insertAfterAtRulesNames.includes(x.name))
            const commentLastIndex = root.nodes.findLastIndex(x => x.type === 'comment' && regExpTest(insertAfterComments, x.text))
            const lastIndex = Math.max(atruleLastIndex, commentLastIndex)
            for (const params of directiveParams) {
              const node = root.nodes.find(x => x.type === 'atrule' && x.params === params)
              if (!node) {
                const atRule = helpers.atRule({ name: 'tailwind', params })
                if (lastIndex < 0) {
                  root.prepend(atRule)
                }
                else {
                  root.insertAfter(lastIndex, atRule)
                }
              }
            }

            const cfg = typeof config === 'function' ? config(root.source?.input) : config
            let tailwindcssConfig: Config | undefined
            if (typeof cfg === 'string') {
              const result = await loadConfig({
                cwd,
                config: cfg,
              })
              if (result) {
                tailwindcssConfig = result.config
              }
            }
            else {
              tailwindcssConfig = (cfg ?? {
                content: [],
              }) as Config
            }

            if (tailwindcssConfig && root.source?.input && root.source.input.file) {
              const basename = removeFileExtension(root.source.input.file)
              set(tailwindcssConfig, 'content', [
                `${basename}.${extensionsGlob}`,
              ])

              // 分析模板
              const deps = await getDepFiles(`${basename}.wxml`)
              for (const dep of deps) {
                if (Array.isArray(tailwindcssConfig.content)) {
                  tailwindcssConfig.content.push(`${removeFileExtension(dep)}.${extensionsGlob}`)
                }
              }

              await postcss([
                tailwindcss(tailwindcssConfig),
              ]).process(root, {
                from: root.source.input.file,
              }).async()
            }
          }
        },
      },
    ],
  }
}
creator.postcss = true

export default creator
