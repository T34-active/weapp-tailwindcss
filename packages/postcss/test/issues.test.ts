import { createStyleHandler } from '@/index'
// import tailwindcss from '@tailwindcss/postcss'
// import fs from 'fs-extra'
import path from 'pathe'
import postcss from 'postcss'

async function generateCss(base: string) {
  const tailwindcss = (await import('@tailwindcss/postcss')).default
  return await postcss([
    tailwindcss({
      base,
    }),
  ])
    .process('@import "weapp-tailwindcss";', {
      from: './index.ts',
    })
}

describe('issues', () => {
  it('https://github.com/sonofmagic/weapp-tailwindcss/issues/631', async () => {
    const code = await generateCss(path.resolve(__dirname, './fixtures/issues/631'))
    expect(code.css).toMatchSnapshot()
    const styleHandler = createStyleHandler({
      isMainChunk: true,
    })
    const { css } = await styleHandler(code.css, {
      isMainChunk: true,
    })
    expect(css).toMatchSnapshot()
  })
  it('https://github.com/sonofmagic/weapp-tailwindcss/issues/632', async () => {
    const code = await generateCss(path.resolve(__dirname, './fixtures/issues/632'))
    expect(code.css).toMatchSnapshot()
    const styleHandler = createStyleHandler({
      isMainChunk: true,
    })
    const { css } = await styleHandler(code.css, {
      isMainChunk: true,
    })
    expect(css).toMatchSnapshot()
  })

  it('https://github.com/sonofmagic/weapp-tailwindcss/issues/638', async () => {
    const code = await generateCss(path.resolve(__dirname, './fixtures/issues/638'))
    expect(code.css).toMatchSnapshot()
    const styleHandler = createStyleHandler({
      isMainChunk: true,
    })
    const { css } = await styleHandler(code.css, {
      isMainChunk: true,
    })
    expect(css).toMatchSnapshot()
  })

  it('https://github.com/sonofmagic/weapp-tailwindcss/issues/638 case 0', async () => {
    const code = await generateCss(path.resolve(__dirname, './fixtures/issues/638-0'))
    expect(code.css).toMatchSnapshot()
    const styleHandler = createStyleHandler({
      isMainChunk: true,
    })
    const { css } = await styleHandler(code.css, {
      isMainChunk: true,
    })
    expect(css).toMatchSnapshot()
  })
})
