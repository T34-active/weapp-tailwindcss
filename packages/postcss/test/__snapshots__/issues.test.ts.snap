// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`issues > https://github.com/sonofmagic/weapp-tailwindcss/issues/631 1`] = `
"/*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */
:root, :host {
  --color-blue-500: oklch(62.3% .214 259.815);
  --spacing: .25rem;
}
*,:after,:before,::backdrop {
  box-sizing: border-box;
  border: 0 solid;
  margin: 0;
  padding: 0;
}
.size-12 {
  width: calc(var(--spacing) * 12);
  height: calc(var(--spacing) * 12);
}
.bg-blue-500\\/30 {
  background-color: color-mix(in srgb, oklch(62.3% .214 259.815) 30%, transparent);
  @supports (color: color-mix(in lab, red, red)) {
    background-color: color-mix(in oklab, var(--color-blue-500) 30%, transparent);
  }
}"
`;

exports[`issues > https://github.com/sonofmagic/weapp-tailwindcss/issues/631 2`] = `
"/*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */
page,:host {
  --color-blue-500: rgb(50, 128, 255);
  --spacing: .25rem;
}
@supports (color: color(display-p3 0 0 0%)) {
page,:host {
  --color-blue-500: rgb(50, 128, 255);
}

@media (color-gamut: p3) {
page,:host {
  --color-blue-500: color(display-p3 0.26642 0.49122 0.98862);
}
}
}
view,text,:after,:before,::backdrop {
  box-sizing: border-box;
  border: 0 solid;
  margin: 0;
  padding: 0;
}
.size-12 {
  width: calc(var(--spacing) * 12);
  height: calc(var(--spacing) * 12);
}
.bg-blue-500s30 {
  background-color: rgba(50, 128, 255, 0.3);
}
@media (color-gamut: p3) {
.bg-blue-500s30 {
  background-color: color(display-p3 0.26642 0.49122 0.98862 / 0.3);
}
}"
`;

exports[`issues > https://github.com/sonofmagic/weapp-tailwindcss/issues/632 1`] = `
"/*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */
:root, :host {
  --color-blue-100: oklch(93.2% .032 255.585);
}
*,:after,:before,::backdrop {
  box-sizing: border-box;
  border: 0 solid;
  margin: 0;
  padding: 0;
}
.hover\\:border-blue-100\\/20 {
  &:hover {
    @media (hover: hover) {
      border-color: color-mix(in srgb, oklch(93.2% .032 255.585) 20%, transparent);
      @supports (color: color-mix(in lab, red, red)) {
        border-color: color-mix(in oklab, var(--color-blue-100) 20%, transparent);
      }
    }
  }
}"
`;

exports[`issues > https://github.com/sonofmagic/weapp-tailwindcss/issues/632 2`] = `
"/*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */
page,:host {
  --color-blue-100: rgb(219, 234, 254);
}
view,text,:after,:before,::backdrop {
  box-sizing: border-box;
  border: 0 solid;
  margin: 0;
  padding: 0;
}
.hovercborder-blue-100s20:hover {
      border-color: rgba(219, 234, 254, 0.2);
  }"
`;
